### PR TITLE
Add convertJSON option

### PR DESCRIPTION
**Pull request by pbelyaev**
#1 

This option converts json if there is such value in a cookie. If you want to use this feature you have to specify it on initialization step, e.g.: 

```javascript
Vue.use(VueReactiveCookie, {
  convertJSON: true,
});
```

### CODE REVIEW

File: src/vue-reactive-cookie.js [hunk 1]
```diff
@@ -1,13 +1,27 @@
 import Cookies from 'js-cookie';
 
 export default {
-  install(Vue) {
+  install(Vue, options) {
     this.Vue = Vue;
+    this.convertJSON = typeof options.convertJSON !== 'undefined' ? options.convertJSON : false;
```
Comment (marcushultman) [inline src/vue-reactive-cookie.js]: What happens if no option object is passed?

File: src/vue-reactive-cookie.js [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+    this.cookies = this.createNewCookieInstance();
```
Comment (marcushultman) [inline src/vue-reactive-cookie.js]: `createNewCookiesInstance`?

File: src/vue-reactive-cookie.js [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
     this.instance = this.createNewVueInstance();
     this.defineVueInstanceProperties();
     this.updateCookiesInstance();
   },
 
+  createNewCookieInstance() {
+    return Cookies.withConverter(value => {
+      const decodedValue = decodeURIComponent(value);
+
+      if (this.convertJSON && this.isJSON(decodedValue)) {
```
Comment (marcushultman) [inline src/vue-reactive-cookie.js]: decodeURIComponent inside the first check?

**Comment by marcushultman** [comment]
Is the `isJSON` implementation taken from somewhere? I guess we could get false positives if we'd simply use `return JSON.parse(decodedValue) || decodedValue;`, correct?

File: src/vue-reactive-cookie.js (hunk 1 above)
Comment (pbelyaev) [reply src/vue-reactive-cookie.js]: Haven't thought about it, will add an empty object as a default value

File: src/vue-reactive-cookie.js (hunk 3 above)
Comment (pbelyaev) [reply src/vue-reactive-cookie.js]: Looks ugly but will works only if it really needs. Nice catch.

File: src/vue-reactive-cookie.js (hunk 2 above)
Comment (pbelyaev) [reply src/vue-reactive-cookie.js]: 👍

**Comment by pbelyaev** [comment]
@marcushultman it has been taken from prototype.js. If we will ```JSON.parse(decodedValue)``` without any checking then we have to use ```try { /* ... */ } catch { /* ... */ }``` because it's gonna return SyntaxError ```Unexpected token a in JSON```

**Comment by pbelyaev** [comment]
@marcushultman lemme know if I forget something. Will merge it tomorrow.

**Review by marcushultman — APPROVED** [review]

**Pull request merged**